albumentations/augmentations/functional.py

```diff
@@ -1964,3 +1965,27 @@ def fancy_pca(img, alpha=0.1):
     orig_img = orig_img.astype(np.uint8)
 
     return orig_img
+
+
+@jit(nopython=True)
+def _shuffle(x, i, h, w, md):
+    dx, dy = np.random.randint(-md, md, size=(2,))
+    h_prime, w_prime = h + dy, w + dx
+    x[h, w], x[h_prime, w_prime] = x[h_prime, w_prime], x[h, w]
+    return x
+
+
+@clipped
+def glass_blur(img, sigma, max_delta, iterations):
+    coef = MAX_VALUES_BY_DTYPE[img.dtype]
+    x = np.uint8(cv2.GaussianBlur(np.array(img) / coef, sigmaX=sigma, ksize=(0, 0)) * coef)
+
+    # locally shuffle pixels
+    for i, h, w in product(
+        range(iterations),
+        range(img.shape[0] - max_delta, max_delta, -1),
+        range(img.shape[1] - max_delta, max_delta, -1),
+    ):
+        x = _shuffle(x, i, h, w, max_delta)
```

Comment:
If you use numba, please implemet `for` exprasion inside numba function, else i don`t see why there need numba.

Reply:
In the latest version I implemented both versions, because the original and the faster numpy version give a bit different result.
So there is a "fast" implementation with the code by taxicoo and an "exact" implementation with the code in numba.
I tried to wrap numba over the loop, but the speed decreased. Having numba over the `_shuffle` function has a huge increase over not having it.